src/Async.js

```diff
@@ -34,14 +42,16 @@ type State = {
   passEmptyOptions: boolean,
 };
 
-export const makeAsyncSelect = (SelectComponent: ComponentType<*>) =>
-  class Async extends Component<Props, State> {
+export const makeAsyncSelect = <C: {}>(
+  SelectComponent: AbstractComponent<C>
+): AbstractComponent<C & AsyncProps> =>
```

Comment:
I would use spread instead

Reply:
spread is a good idea if you have exact flow types